orttraining/orttraining/core/optimizer/compute_optimizer/padding_elimination.cc

```diff
@@ -216,45 +283,23 @@ void IterateSubgraphFromNode(Graph& graph,
         graph_utils::IsSupportedOptypeVersionAndDomain(*cur, "Mul", {7, 13, 14})) {
       ORT_ENFORCE(subgraph.find(cur->MutableInputDefs()[0]) != subgraph.end() ||
                   subgraph.find(cur->MutableInputDefs()[1]) != subgraph.end());
-      NodeArg* arg_in_subgraph = nullptr;
-      NodeArg* arg_not_in_subgraph = nullptr;
-      if (subgraph.find(cur->MutableInputDefs()[0]) != subgraph.end()) {
-        arg_in_subgraph = cur->MutableInputDefs()[0];
-        arg_not_in_subgraph = cur->MutableInputDefs()[1];
-      } else if (subgraph.find(cur->MutableInputDefs()[1]) != subgraph.end()) {
-        arg_in_subgraph = cur->MutableInputDefs()[1];
-        arg_not_in_subgraph = cur->MutableInputDefs()[0];
-      }
-
-      // arg_in_subgraph is contained in subgraph, so its shape must be [batch_size, seq_len, ...]
-      // Now only support cases of the two shapes are absolutely same or the other shape dim size is smaller by 2.
-      // For example, [batch_size, seqlen, hidden_size] and [batch_size, seqlen, hidden_size].
-      //              [batch_size, seqlen, hidden_size] and [hidden_size].
-      // TODO: support other case such as:
-      //              [batch_size, seqlen, hidden_size] and [batch_size, 1, hidden_size]
-      if (arg_in_subgraph->Shape() && arg_not_in_subgraph->Shape() &&
-          (arg_not_in_subgraph->Shape()->dim_size() <= arg_in_subgraph->Shape()->dim_size() - 2 ||
-           (arg_in_subgraph->Shape()->dim_size() == arg_not_in_subgraph->Shape()->dim_size() &&
-            arg_in_subgraph->Shape()->dim(0) == arg_not_in_subgraph->Shape()->dim(0) &&
-            arg_in_subgraph->Shape()->dim(1) == arg_not_in_subgraph->Shape()->dim(1)))) {
+      if (cur->InputDefs()[0]->Shape() && cur->InputDefs()[1]->Shape()) {
+        if ((subgraph.find(cur->MutableInputDefs()[0]) == subgraph.end() &&
```

Comment:
I would suggest not to use mutable unless you really need modify it. 
```suggestion
        if ((subgraph.find(cur->InputDefs()[0]) == subgraph.end() &&
```

Reply:
NodeArgs in subgraph is mutable, So here I must use mutable.

Reply:
yep, but it will report error below if I use MutableInputDefs:
error: invalid conversion from ‘const T*’ {aka ‘const onnxruntime::NodeArg*’} to ‘std::unordered_set<onnxruntime::NodeArg*>::key_type’ {aka ‘onnxruntime::NodeArg*’}